pkg/storage/couchdb/couchdbstore.go

```diff
@@ -311,108 +290,161 @@ func (i *couchDBResultsIterator) Next() (bool, error) {
+		data, err := c.getDataFromAttachment(k)
+		if err != nil {
+			return nil, fmt.Errorf(failureWhileGettingDataFromAttachment, err)
+		}
+
+		return data, nil
 	}
 
 	// Strip out the CouchDB-specific fields
 	delete(rawDoc, "_id")
 	delete(rawDoc, "_rev")
 
-	strippedJSON, err := json.Marshal(rawDoc)
+	strippedJSON, err := c.marshal(rawDoc)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf(failureWhileMarshallingStrippedDoc, err)
 	}
 
 	return strippedJSON, nil
 }
 
-func (c *CouchDBStore) getRevision(k string) (string, error) {
-	rawDoc, err := c.getRawDoc(k)
-	if err != nil {
-		return "", fmt.Errorf(getRevisionFailureErrMsg, err)
-	}
-
-	rev, containsRev := rawDoc["_rev"]
-	if !containsRev {
-		return "", fmt.Errorf(getRevisionFailureErrMsg, errMissingRevisionField)
-	}
-
-	revString, ok := rev.(string)
-	if !ok {
-		return "", fmt.Errorf(getRevisionFailureErrMsg, errFailToAssertRevAsString)
-	}
-
-	return revString, nil
-}
-
 func (c *CouchDBStore) getRawDoc(k string) (map[string]interface{}, error) {
 	rawDoc := make(map[string]interface{})
 
 	row := c.db.Get(context.Background(), k)
 
 	err := row.ScanDoc(&rawDoc)
 	if err != nil {
-		if strings.Contains(err.Error(), couchDBNotFoundErr) {
-			return nil, fmt.Errorf(getRawDocFailureErrMsg, storage.ErrValueNotFound)
+		if strings.Contains(err.Error(), docNotFoundErrMsgFromKivik) {
+			return nil, fmt.Errorf(failureWhileScanningResultRowsDoc, storage.ErrValueNotFound)
 		}
 
-		return nil, fmt.Errorf(getRawDocFailureErrMsg, err)
+		return nil, fmt.Errorf(failureWhileScanningResultRowsDoc, err)
 	}
 
 	return rawDoc, nil
 }
 
+func (c *CouchDBStore) getRevID(k string) (string, error) {
+	rawDoc, err := c.getRawDoc(k)
+	if err != nil {
+		return "", fmt.Errorf(getRawDocFailureErrMsg, err)
+	}
+
+	revID, containsRevID := rawDoc["_rev"]
+	if !containsRevID {
+		return "", errMissingRevIDField
+	}
+
+	revIDString, ok := revID.(string)
+	if !ok {
+		return "", errFailToAssertRevIDAsString
+	}
+
+	return revIDString, nil
+}
+
 func (c *CouchDBStore) getDataFromAttachment(k string) ([]byte, error) {
 	attachment, err := c.db.GetAttachment(context.Background(), k, "data")
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf(failureDuringCouchDBGetAttachmentCall, err)
 	}
 
-	data, err := ioutil.ReadAll(attachment.Content)
+	data, err := c.readAll(attachment.Content)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf(failureWhileReadingAttachmentContent, err)
 	}
 
 	return data, nil
 }
+
+func (c *CouchDBStore) addRevID(valueToPut []byte, revID string) ([]byte, error) {
+	var m map[string]interface{}
+	if err := json.Unmarshal(valueToPut, &m); err != nil {
+		return nil, fmt.Errorf(failureWhileUnmarshallingPutValue, err)
+	}
+
+	m["_rev"] = revID
+
+	newValue, err := c.marshal(m)
+	if err != nil {
+		return nil, fmt.Errorf(failureWhileMarshallingPutValueWithNewlyAddedRevID, err)
+	}
+
+	return newValue, nil
+}
+
+func isJSON(textToCheck []byte) bool {
```

Comment:
you can just check if `strings.HasPrefix(string(textToCheck), "{")`, right?

Reply:
Technically the following are valid JSON but don't have that `{` character:

`true`
`false`

That `json.Unmarshal` call correctly handles those cases